### PR TITLE
[Event Hubs Client] Track Two: Second Preview (Minor Doc Updates)

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

The intent of these changes is add the API reference link into the README and introduce the initial design proposal for preview two.

# Last Upstream Rebase

Wednesday, July 10, 2019  9:36am (EDT)

### CODE REVIEW

**Review by mssfang — APPROVED** [review]

File: sdk/eventhub/Azure.Messaging.EventHubs/design/proposal-event-hubs-second-preview.md [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+#### Creating an Event Hub client with custom retry configuration
+
+```csharp
+var clientOptions = new EventHubClientOptions();
+
+clientOptions.RetryMode = RetryMode.Fixed;
+clientOptions.MaxRetries = 5;
+clientOptions.Delay = TimeSpan.FromMilliseconds(250);
+clientOptions.MaxDelay = TimeSpan.FromSeconds(2);
+clientOptions.TryTimeout = TimeSpan.FromSeconds(90);
```
Comment (conniey) [inline sdk/eventhub/Azure.Messaging.EventHubs/design/proposal-event-hubs-second-preview.md]: `clientOptions.TryTimeout` is an odd name... though, I'm not sure what a better one is.

File: sdk/eventhub/Azure.Messaging.EventHubs/design/proposal-event-hubs-second-preview.md [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+- `OperationCancelledException`
+- `TimeoutException`
+- `MessageSizeExceededException`
+- `QuotaExceededException`
+
+#### `Azure.Messaging.EventHubs.Core`
+
+The location for internal types used by the Event Hubs library to facilitate operations; these constructs are not intended to be consumed externally. 
+
+##### _Example Types:_
```
Comment (conniey) [inline sdk/eventhub/Azure.Messaging.EventHubs/design/proposal-event-hubs-second-preview.md]: I think you were using sentence casing for your headers, so would this be "Example types:"?

File: sdk/eventhub/Azure.Messaging.EventHubs/design/proposal-event-hubs-second-preview.md [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+- `AmqpMessageConverter`
+- `AmqpEventDataBatch`
+
+#### `Azure.Messaging.EventHubs.Diagnostics`
+
+The location for internal types used by the Event Hubs library for diagnostics and logging activities; these constructs are not intended to be consumed externally. 
+
+##### _Example Types:_
+ 
+- `EventHubsEventSource`
```
Comment (conniey) [inline sdk/eventhub/Azure.Messaging.EventHubs/design/proposal-event-hubs-second-preview.md]: Would it be EventHubEventSource? Or is this capable of listening to an entire namespace?

**Review by conniey — APPROVED** [review]
LGTM

File: sdk/eventhub/Azure.Messaging.EventHubs/design/proposal-event-hubs-second-preview.md (hunk 2 above)
Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs/design/proposal-event-hubs-second-preview.md]: Yeah, a bit inconsistent there.   Thanks!

File: sdk/eventhub/Azure.Messaging.EventHubs/design/proposal-event-hubs-second-preview.md [hunk 4]
```diff
@@ -0,0 +1,389 @@
+﻿# .NET Event Hubs Client: Track Two Proposal (Second Preview)
+
+Azure Event Hubs is a highly scalable publish-subscribe service that can ingest millions of events per second and stream them to multiple consumers. This lets you process and analyze the massive amounts of data produced by your connected devices and applications. Once Event Hubs has collected the data, you can retrieve, transform and store it by using any real-time analytics provider or with batching/storage adapters. If you would like to know more about Azure Event Hubs, you may wish to review: [What is Event Hubs](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-about)? 
+
+## Design Overview
+
+This design is focused on the second preview of the track two Event Hubs client library, and limits the scope of discussion to those areas with active development for the second preview. For wider context and more general discussion of the design goals for the track two Event Hubs client, please see the [.NET Event Hubs Client: Track Two Proposal (First  Preview)](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/design/proposal-event-hubs-first-preview.md).
```
[the remark this answers is not in the document]
Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs/design/proposal-event-hubs-second-preview.md]: Odd.  My editor is showing a single space.  Tried adjusting, lets see what hilarity ensues.

File: sdk/eventhub/Azure.Messaging.EventHubs/README.md [hunk 5]
```diff
@@ -12,7 +12,7 @@ The Azure Event Hubs client library allows for publishing and consuming of Azure
 
 - Receive events from one or more publishers, transform them to better meet the needs of your ecosystem, then publish the transformed events to a new stream for consumers to observe.
 
-[Source code](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/eventhub/Azure.Messaging.EventHubs) | [Package (NuGet)](https://www.nuget.org/packages/Azure.Messaging.EventHubs/) | API reference documentation (coming soon) | [Product documentation](https://docs.microsoft.com/en-us/azure/event-hubs/)
+[Source code](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/eventhub/Azure.Messaging.EventHubs) | [Package (NuGet)](https://www.nuget.org/packages/Azure.Messaging.EventHubs/) | [API reference documentation](https://azure.github.io/azure-sdk-for-net/api/EventHubs/Azure.Messaging.EventHubs.html) | [Product documentation](https://docs.microsoft.com/en-us/azure/event-hubs/)
```
[the remark this answers is not in the document]
Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs/README.md]: Nice... I did not know that!

**Pull request merged**